### ISSUE TITLE
"Test events not received" when running JUnit tests with Gradle

### ISSUE THREAD

**Issue by Sahilgat** [nus-cs2103-AY1920S1/forum#39]
![Code](https://user-images.githubusercontent.com/38809578/64129514-197cd500-cdef-11e9-9ab1-e19d79f668cb.PNG)
![Log](https://user-images.githubusercontent.com/38809578/64129523-2994b480-cdef-11e9-9336-e9e182061eb5.PNG)
Hi, when running my JUnit tests from Gradle I get the following output but I'm not sure what to make of it. I've reread the instructions and am fairly certain my set-up is correct and the folder structure is fine as well, is there something else I might be doing wrong? I've done some googling but they suggest other dependency issues and I find the code quite difficult to understand as well as it is usually part of larger projects. I've attached the code of my JUnit tests below was well.
![1](https://user-images.githubusercontent.com/38809578/64129613-bfc8da80-cdef-11e9-8346-b32806c1f5b4.PNG)
![2](https://user-images.githubusercontent.com/38809578/64129615-c35c6180-cdef-11e9-896d-3ad06ef48cfe.PNG)

**Comment by j-lum** [comment]
Have you tried running gradle from the terminal?

**Comment by Sahilgat** [comment]
> Have you tried running gradle from the terminal?

Using the command "gradlew test" from the terminal results in the following output when run from the duke folder (C:\Users\nisga\OneDrive\Desktop\duke):
BUILD SUCCESSFUL in 990ms
3 actionable tasks: 3 up-to-date

Is this expected output or is there something else I am missing? Thanks for your quick response!

**Comment by j-lum** [comment]
That appears to be fine. 

You might want check if your project structure is in order because it seems to be a problem with your IDE config.

**Comment by Sahilgat** [comment]
> That appears to be fine.
> 
> You might want check if your project structure is in order because it seems to be a problem with your IDE config.

Is there a specific way to be checking this? As far as I understand project structure refers to the order of my files but when I select "Tests" from my project menue I am being directed to the correct folder so I believed my project structure is in order as shown in the screenshot below. Did you mean something else by project structure? Thanks again for your help!
![Capture](https://user-images.githubusercontent.com/38809578/64130454-f8b77e00-cdf4-11e9-8afc-d3274aaed1b3.PNG)

**Comment by j-lum** [comment]
I was referring to the `Project Structure...` feature in IntelliJ IDEA. Try this:

1. Enter `gradlew cleanTest` in a terminal.
1. Try running your tests again from IntelliJ.

**Comment by Sahilgat** [comment]
> I was referring to the `Project Structure...` feature in IntelliJ IDEA. Try this:
> 
> 1. Enter `gradlew cleanTest` in a terminal.
> 2. Try running your tests again from IntelliJ.

Thanks so much for your help! I think that indeed was the issue! I'll try moving the folders and restructuring the project accordingly. Also ran the gradlew cleanTest and output was: 

BUILD SUCCESSFUL in 896ms
1 actionable task: 1 executed

![Capture](https://user-images.githubusercontent.com/38809578/64130763-01a94f00-cdf7-11e9-8231-44f901e1eb00.PNG)

**Comment by j-lum** [comment]
Technically, that's not a problem. Gradle just recognized that there were no new changes and skipped testing. As a result, no results were reported to IntelliJ. By running `:cleanTest`, you basically forced Gradle to compile and run the tests again.

**Comment by Sahilgat** [comment]
> Technically, that's not a problem. Gradle just recognized that there were no new changes and skipped testing. As a result, no results were reported to IntelliJ. By running `:cleanTest`, you basically forced Gradle to compile and run the tests again.

Yea I just realised taking a closer look that the folder structure doesn't seem to be an issue. However, I still get the "Test events were not recieved" original error even after running gradlew cleanTest and can't see the output of the tests passing as shown in the examples I've seen. Am I correct in assuming my tests work for now though and is there anything I need to change to get the same output as shown below? Or am I correct in assuming everything is fine and the result shown in the example below is because Gradle was not used?
![Capture](https://user-images.githubusercontent.com/38809578/64130968-3d90e400-cdf8-11e9-827a-8893254ee895.PNG)

**Comment by j-lum** [comment]
You can change the test runner to IntelliJ in the settings.
![image](https://user-images.githubusercontent.com/22460123/64131056-b7c16880-cdf8-11e9-864a-0dd381a8a636.png)

**Comment by Sahilgat** [comment]
> You can change the test runner to IntelliJ in the settings.
> ![image](https://user-images.githubusercontent.com/22460123/64131056-b7c16880-cdf8-11e9-864a-0dd381a8a636.png)

Ok thanks again for your help I think I more or less figured it out! One last question, when I swapped to IntelliJ IDEA and then selected "Add JUnit 5.* to classpath" under Test as per the instructions under Week 3, my test files now get errors highlighted and when I try to run all tests I get the error:
"No tests were found" and the popup "Tests were not found in module "test". Use module "main" instead" as shown in the screenshot below. When I switched back to Gradle the tests still ran fine. Is there any way to configure it such that I can switch between both easily? I suspect the adding JUnit to classpath was what caused the error as simply swapping to using IntelliJ in the settings did not cause the issue at first. Else it seems I'll stick to using Gradle though I want to understand what's happening here.
![Capture](https://user-images.githubusercontent.com/38809578/64131816-c3fbf480-cdfd-11e9-9687-cded4a55d510.PNG)
